### DISCUSSION TITLE
debugpy and python version

### DISCUSSION THREAD

**Discussion by bluss** [microsoft/debugpy#1574] [Q&A] [answered]
Maybe some clarifications are needed, we can install debugpy separately in its own isolated virtualenv, where debugpy and the application being debugged are in separate virtualenvs (is this right?)

Are there any requirements on python versions, do they need to use the same python version or can it be different?

**Comment by int19h** [comment]
debugpy itself doesn't have any dependencies (that it doesn't vendor), so there's technically no need for it to be installed into any environment - you can even just clone the repo and then do `python .../src/debugpy` directly. 

However, if you *do* pip install it into an environment, that environment would need to be active for you to be able to do `python -m debugpy ...`. Whichever script or module you pass to debugpy as command-line argument is started using `runpy` standard module, so it will also run in that environment.

**Reply by bluss** [reply]
I forgot some crucial context - I'm sorry :man_facepalming:  -- but thanks for the info!

In the context of this debugger's use of debugpy https://github.com/mfussenegger/nvim-dap-python?tab=readme-ov-file#usage they seem to suggest it will work correctly with application and debugpy in separate virtualenvs, and it actually seems to work that way(?), but what about separate python versions in those enivronments?

**Reply by int19h** [reply, marked as answer]
Ah, I see. Yes, it is fine to have separate Python versions in this case.
